### PR TITLE
Upgrade Gradle to 7.2

### PR DESCRIPTION
**Pull request by pbielicki**
Use VERSION_CATALOGS feature
Use JUnit Platform + Vintage

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+[versions]
+spock = "2.0-groovy-3.0"
+
+[libraries]
+asciidoctorj = "org.asciidoctor:asciidoctorj:1.5.8.1"
+cglib = "cglib:cglib-nodep:3.3.0"
+commons-io = "commons-io:commons-io:2.4"
+commons-lang3 = "org.apache.commons:commons-lang3:3.7"
+groovy = "org.codehaus.groovy:groovy:3.0.8"
+junit = "junit:junit:4.12"
```
Comment (marcphilipp) [inline gradle/libs.versions.toml:10]:
```suggestion
junit = "junit:junit:4.13.2"
```

File: samples-check/src/test/samples/gradle/composite-sample/basic/compositeBuildsBasicCli.sample.out [hunk 2]
```diff
@@ -13,6 +13,5 @@
 > Task :run
 The answer is 42
 
-
 BUILD SUCCESSFUL in 0s
-2 actionable tasks: 2 executed
+6 actionable tasks: 6 executed
```
Comment (marcphilipp) [inline samples-check/src/test/samples/gradle/composite-sample/basic/compositeBuildsBasicCli.sample.out:17]: Where included builds not counted previously?

**Review by marcphilipp — APPROVED** [review]

**Review by britter — APPROVED** [review]
LGTM

File: samples-check/src/test/samples/gradle/composite-sample/basic/compositeBuildsBasicCli.sample.out (hunk 2 above)
Comment (pbielicki) [reply samples-check/src/test/samples/gradle/composite-sample/basic/compositeBuildsBasicCli.sample.out:17]: it looks like - shall I investigate it or we're good?

File: samples-check/build.gradle.kts [hunk 3]
```diff
@@ -1,27 +1,27 @@
...
+    api(libs.junit)
+    compileOnly(libs.jsr305)
+    implementation(libs.commons.io)
+    implementation(libs.commons.lang3)
     implementation(gradleTestKit())
-    testImplementation(Libraries.SPOCK_CORE)
-    testImplementation(Libraries.CGLIB)
-    testImplementation(Libraries.OBJENESIS)
-    testImplementation(Libraries.GROOVY)
+    testImplementation(libs.cglib)
```
[the remark this answers is not in the document]
Comment (pbielicki) [reply samples-check/build.gradle.kts:13]: good catch - I was sure I checked it but it was only my fantasy
I removed this directly on `master` branch now

File: samples-check/src/test/samples/gradle/composite-sample/basic/my-utils/build.gradle [hunk 4]
```diff
@@ -14,6 +14,6 @@ subprojects {
 
 project(":string-utils") {
     dependencies {
-        compile "org.apache.commons:commons-lang3:3.4"
+        implementation "org.apache.commons:commons-lang3:3.4"
```
[the remark this answers is not in the document]
Comment (pbielicki) [reply samples-check/src/test/samples/gradle/composite-sample/basic/my-utils/build.gradle:17]: it's a test code so it's not that important

[the remark this answers is not in the document]
Comment (pbielicki) [reply samples-check/src/test/samples/gradle/composite-sample/basic/my-utils/build.gradle:17]: anyway I pushed a change to `master`

**Pull request merged**